bytecode-verificator/bytecode_verificator.sh

```diff
@@ -0,0 +1,479 @@
+
+  echo "Starting local fork ${local_fork_command}"
+  (nc -vz 127.0.0.1 $local_rpc_port) &>/dev/null && kill -15 "$(lsof -t -i:$local_rpc_port)"
+
+  $local_fork_command 1> ./logs 2>& 1 &
+  fork_pid=$$
+  echo "Ganache pid $fork_pid"
+
+  sleep 10
+}
+
+function encode_address() {
+  echo "${1/0x/000000000000000000000000}"
+}
+
+function encode_uint256() {
+  printf "%064X\n" "$1"
+}
+
+function encode_bytes32() {
+  echo "${1/0x/}"
+}
+
+function encode_bytes() {
+  local bytes_str
+  local data_length
+  local encoded_length
+
+  bytes_str=$(sed -E 's/^0x(00)*//' <<< "$1")
+  data_length=$((${#bytes_str} / 2))
+  encoded_length=0
+  if [[ data_length -gt "0" ]]; then
+    encoded_length=$(bc <<< "(((${#bytes_str} - 1) / 64) + 1) * 64")
+  fi
+  bytes_str=$bytes_str$zero_padding
+  bytes_str=${bytes_str:0:$encoded_length}
+  echo "$(printf "%064X\n" "$data_length")$bytes_str"
+}
+
+function encode_string() {
+  local string_bytes
+  local data_length
+  local encoded_length
+
+  string_bytes=$(xxd -p <<< "$1" | sed 's/..$//')
+  data_length=$(bc <<< "${#string_bytes} / 2")
+  encoded_length=0
+  if [[ data_length -gt "0" ]]; then
+    encoded_length=$(bc <<< "(((${#string_bytes} - 1) / 64) + 1) * 64")
+  fi
+  string_bytes=$string_bytes$zero_padding
+  string_bytes=${string_bytes:0:$encoded_length}
+  echo "$(printf "%064X\n" "$data_length")$string_bytes"
+}
+
+function encode_array() {
+  local types=$1
+  local array=$2
+  local array_length
+  local encoded_data
+
+  array_length=$(jq -r 'length' <<< "$array")
+  encoded_data="$encoded_data$(encode_uint256 "$(bc <<< "$array_length * 32")")"
+  if [[ $array_length -gt 0 ]]; then
+    for i in $(seq 0 "$(bc <<< "$array_length - 1")" ); do
+      case $types in
+        address\[\]) encoded_data="$encoded_data$(encode_address "$(jq -r ".[$i]" <<< "$array")")" ;;
+        uint256\[\]) encoded_data="$encoded_data$(encode_uint256 "$(jq -r ".[$i]" <<< "$array")")" ;;
+        bytes32\[\]) encoded_data="$encoded_data$(encode_bytes32 "$(jq -r ".[$i]" <<< "$array")")" ;;
+        *) _err "Unknown constructor argument type '$1', use --constructor-calldata instead" ;;
+      esac
+    done
+  fi
+  echo "$encoded_data"
+}
+
+function encode_tuple() {
+  local types=$1
+  local args=$2
+  local args_length
+  local encoded_data
+
+  args_length=$(jq -r 'length' <<< "$types")
+
+  for arg_index in $(seq 0 "$(bc <<< "$args_length - 1")" ); do
+    local arg_type
+    local arg
+
+    arg_type=$(jq -r ".[$arg_index]" <<< "$types")
+    arg=$(jq -r ".[$arg_index]" <<< "$args")
+
+    case $arg_type in
+      address) encoded_data="$encoded_data$(encode_address "$arg")" ;;
+      uint256) encoded_data="$encoded_data$(encode_uint256 "$arg")" ;;
+      bytes32) encoded_data="$encoded_data$(encode_bytes32 "$arg")" ;;
+      *) _err "Unknown constructor argument type '$arg_type', use --constructor-calldata instead" ;;
+    esac
+  done
+  echo "$encoded_data"
+}
+
+function endode_placeholder() {
```

Comment:
```suggestion
function endode_solidity_lib_placeholder() {
```
or smth similar, just to be precise.

Reply:
Maybe I don't understand the function correctly.